src/pip/_internal/req/req_uninstall.py

```diff
@@ -364,19 +364,28 @@ def remove(self, auto_confirm: bool = False, verbose: bool = False) -> None:
         logger.info("Uninstalling %s:", dist_name_version)
 
         with indent_log():
-            if auto_confirm or self._allowed_to_proceed(verbose):
-                moved = self._moved_paths
+            try:
+                if auto_confirm or self._allowed_to_proceed(verbose):
+                    moved = self._moved_paths
+
+                    for_rename = compress_for_rename(self._paths)
 
-                for_rename = compress_for_rename(self._paths)
+                    for path in sorted(compact(for_rename)):
+                        moved.stash(path)
+                        logger.verbose("Removing file or directory %s", path)
 
-                for path in sorted(compact(for_rename)):
-                    moved.stash(path)
-                    logger.verbose("Removing file or directory %s", path)
+                    for pth in self._pth.values():
+                        pth.remove()
 
-                for pth in self._pth.values():
-                    pth.remove()
+                    logger.info("Successfully uninstalled %s", dist_name_version)
+            except RuntimeError as runerror:
```

Comment:
This try...except covers *far* too large a block of code.

As a general rule, we do not catch errors caused by the stdout/stderr streams being broken. Any catching of exceptions here should be very tightly targetted at dealing with a specific problem. I don't think we know yet what the underlying problem is, so I think this PR is premature.